spec/requests/api/container_deployments_spec.rb

```diff
@@ -0,0 +1,16 @@
+describe ApiController do
+  it "supports collect-data with GET" do
+    allow_any_instance_of(ContainerDeploymentService).to receive(:cloud_init_template_id).and_return(1)
+    api_basic_authorize collection_action_identifier(:container_deployments, :read, :get)
+    run_get container_deployments_url + "/container_deployment_data"
+    expect_request_success
+    expect_hash_to_have_only_keys(response_hash["data"], %w(providers cloud_init_template_id provision))
+  end
+
+  it "creates container deployment with POST" do
+    allow_any_instance_of(ContainerDeployment).to receive(:create_deployment).and_return(true)
+    api_basic_authorize collection_action_identifier(:container_deployments, :create)
+    run_post(container_deployments_url, gen_request(:create, :example_data => true))
+    expect_request_success
+  end
+end
```

Comment:
that's fine for now, when refactoring after the other PR is merged, remove both allow_any_instance_of from here.  Thanks.